contracts/peripheral/YieldStreamer.sol

```diff
@@ -0,0 +1,447 @@
+            address recipient = recipientInfo[id_].recipientAddress;
+            uint256 unclaimedDai = recipientInfo[id_].unclaimedDai;
+            (uint256 principal, uint256 totalGOHM) = _closeDeposit(id_);
+            delete recipientInfo[id_];
+
+            for (uint256 i = 0; i < activeDepositIds.length; i++) {
+                // Delete integer from array by swapping with last element and calling pop()
+                if (activeDepositIds[i] == id_) {
+                    activeDepositIds[i] = activeDepositIds[activeDepositIds.length - 1];
+                    activeDepositIds.pop();
+                    break;
+                }
+            }
+
+            uint256[] storage recipientIdsArray = recipientIds[recipient];
+            for (uint256 i = 0; i < recipientIdsArray.length; i++) {
+                // Delete integer from array by swapping with last element and calling pop()
+                if (recipientIdsArray[i] == id_) {
+                    recipientIdsArray[i] = recipientIdsArray[recipientIdsArray.length - 1];
+                    recipientIdsArray.pop();
+                    break;
+                }
+            }
+
+            IERC20(gOHM).safeTransfer(msg.sender, principal);
+            IERC20(gOHM).safeTransfer(recipient, totalGOHM - principal);
+            if (unclaimedDai != 0) {
+                IERC20(DAI).safeTransfer(recipient, unclaimedDai);
+            }
+        } else {
+            _withdrawPrincipal(id_, amount_);
+            IERC20(gOHM).safeTransfer(msg.sender, amount_);
+        }
+
+        emit Withdrawn(msg.sender, amount_);
+    }
+
+    /**
+        @notice Withdraw excess yield from your deposit in gOHM.
+        @dev  Use withdrawYieldAsDai() to withdraw yield as DAI.
+        @param id_ Id of the deposit.
+    */
+    function withdrawYield(uint256 id_) external override {
+        if (withdrawDisabled) revert WithdrawDisabled();
+        if (depositInfo[id_].depositor != msg.sender && recipientInfo[id_].recipientAddress != msg.sender)
+            revert UnauthorisedAction();
+
+        recipientInfo[id_].lastUpkeepTimestamp = block.timestamp;
+
+        uint256 yield = _redeemYield(id_);
+
+        IERC20(gOHM).safeTransfer(msg.sender, yield);
+    }
+
+    /**
+        @notice Withdraw excess yield from your deposit in DAI
+        @param id_ Id of the deposit
+    */
+    function withdrawYieldAsDai(uint256 id_) external override {
+        if (withdrawDisabled) revert WithdrawDisabled();
+        if (depositInfo[id_].depositor != msg.sender && recipientInfo[id_].recipientAddress != msg.sender)
+            revert UnauthorisedAction();
+
+        recipientInfo[id_].lastUpkeepTimestamp = block.timestamp;
+
+        uint256 gOHMYield = _redeemYield(id_);
+        uint256 totalOhmToSwap = staking.unwrap(address(this), gOHMYield);
+        staking.unstake(address(this), totalOhmToSwap, false, false);
+
+        IERC20(OHM).approve(address(sushiRouter), totalOhmToSwap);
```

Comment:
Use SafeERC20::approve?  Also might want to consider doing this in the constructor for large amount (0xFFFFFF) to save gas so doesn't need to be performed each time